pkg/network/tracer/offsetguess_test.go

```diff
@@ -299,3 +301,27 @@ func testOffsetGuess(t *testing.T) {
 		assert.Equal(t, offset, consts[o], "unexpected offset for %s", o)
 	}
 }
+
+func TestOffsetGuessPortIPv6Overlap(t *testing.T) {
+	ebpftest.TestBuildMode(t, ebpftest.RuntimeCompiled, "", func(t *testing.T) {
+		addrs, err := offsetguess.GetIPv6LinkLocalAddress()
+		require.NoError(t, err)
+
+		// force preference for the added prefix. 0x35 (53) = DNS port
+		const portMatchingPrefix = "fe80::35:"
+		oldPrefix := offsetguess.IPv6LinkLocalPrefix
+		t.Cleanup(func() { offsetguess.IPv6LinkLocalPrefix = oldPrefix })
+		offsetguess.IPv6LinkLocalPrefix = portMatchingPrefix
+
+		// add IPv6 link-local addresses with 0x35 (53) bytes to each interface
+		for i, addr := range addrs {
+			_, err := nettestutil.RunCommand(fmt.Sprintf("ip -6 addr add %s%d/64 dev %s scope link", portMatchingPrefix, i+1, addr.Zone))
+			require.NoError(t, err)
```

Comment:
Do we want to fail here if there is an address conflict, or continue?

Reply:
> Do we want to fail here if there is an address conflict, or continue?

Meaning the link-local address is already the one we are trying to use? AFAIK that isn't really possible because most implementations use either the MAC or a random ID for a bunch of the bits.